.gitlab/stream-ci.yaml

```diff
@@ -0,0 +1,376 @@
+        BUILD_OPENGL: "ON"
+      - C_COMPILER: gcc-7
+        CXX_COMPILER: g++-7
+        BUILD_SAMPLES: "ON"
+        BUILD_OPENGL: "OFF"
+      - C_COMPILER: gcc-8
+        CXX_COMPILER: g++-8
+        BUILD_SAMPLES: "ON"
+        BUILD_OPENGL: "ON"
+      - C_COMPILER: clang-8
+        CXX_COMPILER: clang++-8
+        BUILD_SAMPLES: ["ON", "OFF"]
+        BUILD_OPENGL: "ON"
+      - C_COMPILER: clang-8
+        CXX_COMPILER: clang++-8
+        BUILD_SAMPLES: "ON"
+        BUILD_OPENGL: "OFF"
+      - C_COMPILER: clang-9
+        CXX_COMPILER: clang++-9
+        BUILD_SAMPLES: "ON"
+        BUILD_OPENGL: "ON"
+      - C_COMPILER: clang-10
+        CXX_COMPILER: clang++-10
+        BUILD_SAMPLES: "ON"
+        BUILD_OPENGL: "ON"
+  before_script:
+    - export PATH=/opt/Kitware/CMake/$CMAKE_MINIMUM/bin:$PATH
+
+.exec-matrix-latest:
+  parallel:
+    matrix:
+  # NOTE: Cutting edge CMake versions will likely not use old compilers
+      - C_COMPILER: gcc-9
+        CXX_COMPILER: g++-9
+        BUILD_SAMPLES: "ON"
+        BUILD_OPENGL: "ON"
+      - C_COMPILER: gcc-10
+        CXX_COMPILER: g++-10
+        BUILD_SAMPLES: ["ON", "OFF"]
+        BUILD_OPENGL: "ON"
+      - C_COMPILER: gcc-10
+        CXX_COMPILER: g++-10
+        BUILD_SAMPLES: "ON"
+        BUILD_OPENGL: "OFF"
+      - C_COMPILER: clang-11
+        CXX_COMPILER: clang++-11
+        BUILD_SAMPLES: "ON"
+        BUILD_OPENGL: "ON"
+      - C_COMPILER: clang-12
+        CXX_COMPILER: clang++-12
+        BUILD_SAMPLES: "ON"
+        BUILD_OPENGL: "ON"
+      - C_COMPILER: clang-13
+        CXX_COMPILER: clang++-13
+        BUILD_SAMPLES: ["ON", "OFF"]
+        BUILD_OPENGL: "ON"
+      - C_COMPILER: clang-13
+        CXX_COMPILER: clang++-13
+        BUILD_SAMPLES: "ON"
+        BUILD_OPENGL: "OFF"
+  before_script:
+    - export PATH=/opt/Kitware/CMake/$CMAKE_LATEST/bin:$PATH
+
+#################
+##             ##
+## Build stage ##
+##             ##
+#################
+
+build:cmake-minimum:
+  extends: .toolchain-matrix-minimum
+  stage: build
+  # NOTE 1: CMake creates build and install folders as needed.
+  # NOTE 2: Lack of space in -H<folder> -B<folder> is important!
+  #         https://stackoverflow.com/a/20611964/1476661
+  script:
+    # When not building OpenGL samples, delete OpenGL-related deps from image to make sure we
+    # don't accidentally include a header that otherwise need not be installed
+    - if [[ "$BUILD_OPENGL" == "OFF" ]]; then /opt/Microsoft/vcpkg/vcpkg remove sfml glm; fi;
+    # Clone, configure, build, install OpenCL-SDK
+    - cmake
+      -G Ninja
+      -D CMAKE_TOOLCHAIN_FILE=/opt/Microsoft/vcpkg/scripts/buildsystems/vcpkg.cmake
+      -D CMAKE_C_COMPILER=$C_COMPILER
+      -D CMAKE_CXX_COMPILER=$CXX_COMPILER
+      -D CMAKE_BUILD_TYPE=Release
+      -D BUILD_TESTING=ON
+      -D ENABLE_OPENCL_LAYERS=ON
+      -D OPENCL_ICD_LOADER_BUILD_TESTING=ON
+      -D BUILD_DOCS=OFF
+      -D BUILD_EXAMPLES=OFF
+      -D BUILD_TESTS=ON
```

Comment:
Should some of these be set in the CMakeLists.txt?

This line in particular I had to set to get anything to build (I suspect that is actually a bug in OpenCL-CLHPP CMake, but it wasn't immediately clear why -- I'll look again later).

Other variables are not essential, but setting them saves building unnecessary tests.

Reply:
Yes, I was thinking hard about making that also part of these PRs, but... scope creep. I really want to make the test option variables consistent. Currently only the ICD-Loader is able to selectively disable it's tests via [`OPENCL_ICD_LOADER_BUILD_TESTING`](https://github.com/KhronosGroup/OpenCL-ICD-Loader/blob/master/CMakeLists.txt#L136-L141) and that's something the CLHPP repo should enable as well. One may want to test the SDK samples, but not build the few hundred header tests. I'll see to it shortly after the SDK and the base changes go through.